copick_live/utils/copick_dataset.py

```diff
@@ -0,0 +1,295 @@
+            )
+            self._point_types[self.current_point] = obj_name
+            if self.current_point not in self._picked_id_per_obj[obj_name]:
+                self._picked_id_per_obj[obj_name].append(self.current_point)
+                self._picked_points_per_obj[obj_name].append(self.current_point_obj)
+            self._store_points(obj_name)
+
+    def handle_reject(self, enable_log=True):
+        if enable_log:
+            self.log_operation(operation="reject", old_obj_name="NC", new_obj_name="NC")
+
+        if self.current_point is not None:
+            self.picked_points_mask[self.current_point] = 2
+            try:
+                obj_name = self._point_types[self.current_point]
+                index = self._picked_id_per_obj[obj_name].index(self.current_point)
+                print(
+                    f"reject point index {self.current_point}, index in the list {index}"
+                )
+                self._picked_id_per_obj[obj_name].pop(index)
+                self._picked_points_per_obj[obj_name].pop(index)
+                self._store_points(obj_name)
+            except:
+                pass
+
+    def handle_assign(self, new_bj_name=None):
+        self.handle_reject(enable_log=False)
+        self.change_obj_name(new_bj_name)
+        self.handle_accept()
+        # self.picked_points_mask[self.current_point] = 3
+
+        # EXPERIMENT, dangeraous, may incurr index errors!
+        # Only re-assignment changes the original states (initialized when load run)
+        new_list = []
+        target = tuple()
+        old_obj_name = self.pickable_obj_name
+        print(old_obj_name, self.points_per_obj[old_obj_name])
+        for item in self.points_per_obj[old_obj_name]:
+            if item[0] != self.current_point:
+                new_list.append(item)
+            else:
+                target = item
+
+        self.points_per_obj[old_obj_name] = new_list
+        # add the new assigned point to the front
+        self.points_per_obj[new_bj_name].insert(0, target)
+
+    def handle_accept_batch(self, point_ids=None, obj_name=None):
+        if point_ids is not None:
+            for point_id in point_ids:
+                self.load_curr_point(point_id=point_id, obj_name=obj_name)
+                self.handle_accept()
+
+    def handle_reject_batch(self, point_ids=None, obj_name=None):
+        if point_ids is not None:
+            for point_id in point_ids:
+                self.load_curr_point(point_id=point_id, obj_name=obj_name)
+                self.handle_reject()
+
+    def handle_assign_batch(self, point_ids=None, obj_name=None, new_bj_name=None):
+        if point_ids is not None and obj_name is not None and new_bj_name is not None:
+            for point_id in point_ids:
+                self.load_curr_point(point_id=point_id, obj_name=obj_name)
+                self.handle_assign(new_bj_name)
+
+
+copick_dataset = None
+
+
+def get_copick_dataset(config_path=None):
```

Comment:
Found another bug: the `counter_checkpoint_file_path` is not passed correctly. Can we define another function that return this path, similar to returning `local_dataset`.

```
def get_local_dataset(config_path=None):
    global local_dataset

    if config_path or not local_dataset:
        config = configparser.ConfigParser()

        if config_path:
            config_path = os.path.abspath(config_path)
        else:
            config_path = os.path.join(os.getcwd(), "config.ini")

        if os.path.exists(config_path):
            config.read(config_path)
        else:
            raise FileNotFoundError(f"Config file not found at {config_path}")

        LOCAL_FILE_PATH = config.get("local_picks", "PICK_FILE_PATH", fallback=None)
        if LOCAL_FILE_PATH:
            LOCAL_FILE_PATH = os.path.join(LOCAL_FILE_PATH, "ExperimentRuns/")
        COPICK_TEMPLATE_PATH = config.get("copick_template", "COPICK_TEMPLATE_PATH", fallback=None)

        if not LOCAL_FILE_PATH or not COPICK_TEMPLATE_PATH:
            raise ValueError("Config paths for LocalDataset are not provided and not found in the config file.")

        local_dataset = LocalDataset(local_file_path=LOCAL_FILE_PATH, config_path=COPICK_TEMPLATE_PATH)

    return local_dataset


def get_counter_file_path(config_path=None):
    global COUNTER_FILE_PATH

    if config_path or not COUNTER_FILE_PATH:
        config = configparser.ConfigParser()

        if config_path:
            config_path = os.path.abspath(config_path)
        else:
            config_path = os.path.join(os.getcwd(), "config.ini")

        if os.path.exists(config_path):
            config.read(config_path)
        else:
            raise FileNotFoundError(f"Config file not found at {config_path}")
    
        COUNTER_FILE_PATH = config.get("counter_checkpoint", "COUNTER_FILE_PATH", fallback=None)

        if not COUNTER_FILE_PATH:
            raise ValueError("Config path for COUNTER_FILE_PATH is not provided and not found in the config file.")
    
    return COUNTER_FILE_PATH